cmd/object-handlers_test.go

```diff
@@ -620,7 +621,7 @@ func testAPIPutObjectStreamSigV4Handler(obj ObjectLayer, instanceType, bucketNam
 			dataLen:            1024,
 			chunkSize:          1024,
 			expectedContent:    []byte{},
-			expectedRespStatus: http.StatusBadRequest,
+			expectedRespStatus: http.StatusInternalServerError,
```

Comment:
We shouldn't change this error, we need to see why we send http.StatusInternalServerError

Reply:
Here in PutObject Handler , we give the higher priority to the compression reader errors ( while compressing from hashReader ). So that we can cover all the  `signatureMismatch`, `malformedEncoding` errors,
So, the copy here 
https://github.com/Praveenrajmani/minio/blob/43572a6d7488471bba7b80166fce0e4f6bb74e50/cmd/object-handlers.go#L934
will respond with `InternalServerError` for an unexpected EOF in case if the content is truncated as per the paticular test case `TestAPIPutObjectStreamSigV4Handler` (Test case - 8)

so here shall i mask the Internal Server Error and reply back with BadRequest to match the test case ?

Reply:
UexpectedEOF should IncompleteBody @Praveenrajmani

Reply:
okay . will add it

Reply:
Done! @harshavardhana